### ISSUE TITLE
Support Haskell2010 as pragma

### ISSUE THREAD

**Issue by jan-christiansen** [kowainik/extensions#106]
`Haskell2010` is an official GHC extension (https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/table.html) and used in [`yi-editor/yi`](https://github.com/yi-editor/yi) (https://github.com/yi-editor/yi/blob/13fa0f7c598a115485572b66fed5bc3cf383d0ef/yi-core/src/Yi/PersistentState.hs#L5), for example. I suggest to support this extension by replacing

```haskell
readExtension :: String -> Maybe Extension
```
with

```haskell
readExtension :: String -> [Extension]
```

and yield `default2010Extensions` in the case of `Haskell2010`.

If this is fine, I will create a pull request.

**Comment by tomjaguarpaw** [comment]
Thanks. Perhaps we do have to do something about that, but I am somewhat surprised that this feature doesn't exist already. The authors would have been aware of `Haskell2010`. That makes me think there is a good reason it is omitted.

**Comment by jan-christiansen** [comment]
Good point, perhaps it is because languages are handled orthogonally to pragmas by Cabal and my approach is rather a hack to make the parser succeed : ) You are right, in my approach the parser would actually successfully parse `NoHaskell2010` : ) As this should obviously not be allowed, adding support for language pragmas would require a larger refactor and explicitly handling language extensions.
